### PR TITLE
Task 2 from thread is done.

### PR DESCRIPTION
**Pull request by qnislavnachev**
Here we have 2 thread, which have to count to different value with different sleep time and when 1 of them finish to  kill other one.

### CODE REVIEW

File: src/main/java/task2/Threads.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            try {
+                sleep(sleepTime);
+            } catch (InterruptedException e) {
+                System.out.println(threadName + " was interrupt.");
+                return;
+            }
+            counter++;
+            System.out.println(threadName + ": " + counter);
+        }
+        thread.interrupt();
```
Comment (tsony-tsonev) [inline src/main/java/task2/Threads.java]: this is not needed

File: src/main/java/task2/Threads.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private int counter = 0;
+    private Threads thread;
+
+    public Threads(String threadName, int countTo, int sleepTime) {
+        this.threadName = threadName;
+        this.countTo = countTo;
+        this.sleepTime = sleepTime;
+    }
+
+    public void hasToInterrupt(Threads thread) {
```
Comment (tsony-tsonev) [inline src/main/java/task2/Threads.java]: just `interrupts` is enough

File: src/main/java/task2/Threads.java (hunk 1 above)
Comment (tsony-tsonev) [inline src/main/java/task2/Threads.java]: By not needed I mean you should call the method of the thread which stops it. The stop method itself should not call interrupt cause this way we will go in the catch clause. It is better to some how make the `run` method of the thread to finish (Most often you just add one more boolean flag to the while cycle).

Comment (qnislavnachev) [inline src/main/java/task2/Threads.java]: fixed

File: src/main/java/task2/Threads.java (hunk 2 above)
Comment (qnislavnachev) [inline src/main/java/task2/Threads.java]: fixed

File: src/main/java/task2/InterruptingThreads.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        this.sleepTime = sleepTime;
+    }
+
+    public void interrupt(InterruptingThreads thread) {
+        this.threadToBeInterrupted = thread;
+    }
+
+    @Override
+    public void run() {
+        while (counter < countTo && flag) {
```
Comment (tsony-tsonev) [inline src/main/java/task2/InterruptingThreads.java]: `||` not  `&&` you want to finish the thread if counter is reached or if thread should be stopped

**Comment by tsony-tsonev** [comment]
LGTM after you fix the last comment

File: src/main/java/task2/InterruptingThreads.java (hunk 3 above)
Comment (qnislavnachev) [inline src/main/java/task2/InterruptingThreads.java]: yes my mistake you are right

Comment (qnislavnachev) [inline src/main/java/task2/InterruptingThreads.java]: fixed

**Pull request merged**